x/merkledb/db.go

```diff
@@ -1006,9 +1009,13 @@ func (db *merkleDB) VerifyChangeProof(
 		// If [proof] has key-value pairs, we should insert children
 		// greater than [end] to ancestors of the node containing [end]
 		// so that we get the expected root ID.
-		largestKey = proof.KeyChanges[len(proof.KeyChanges)-1].Key
+		largestKey = Some(proof.KeyChanges[len(proof.KeyChanges)-1].Key)
+	}
+
+	var largestPath path
+	if largestKey.hasValue {
+		largestPath = newPath(largestKey.value)
```

Comment:
Again, we don't _need_ the `if` here because of `Maybe`'s invariant. What do other people think?